lib/ruby_lsp/requests/semantic_highlighting.rb

```diff
@@ -89,9 +101,10 @@ def visit_vcall(node)
         add_token(node.value.location, :method)
       end
 
-      def add_token(location, type)
+      def add_token(location, type, modifiers = [])
         length = location.end_char - location.start_char
-        @tokens.push(SemanticToken.new(location, length, TOKEN_TYPES.index(type), 0))
+        modifiers_indices = modifiers.map { |modifier| TOKEN_MODIFIERS[modifier] }
```

Comment:
This assumes that the modifier we pass will exist in the `TOKEN_MODIFIERS` hash map. Currently it returns `nil` if the key doesn't exist. We should probably silently warn if the key doesn't exist.

Reply:
I think that'll only happen when VSCode updates modifier names, since all 10 digits are all taken so there won't be new ones? Just curious.
But yeah it'll be helpful for debugging if such situation is logged. I wonder if there's a way to get notified whenever the specification changed though 🤔 

Debugger's [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/specification) has a [json schema](https://microsoft.github.io/debug-adapter-protocol/debugAdapterProtocol.json) that we can use to validate the format in tests. I can't find anything similar for LSP though.

Reply:
Let's change `map` to `filter_map` to automatically ignore modifiers we don't know about and avoid having to handle `nil`.